### PR TITLE
Review the Authentication API for comments

### PR DESCRIPTION
**Pull request by HalynaDumych**
Reviewed and modified the existing comments, added the missing comments.

Relates-to: OLPEDGE-859

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h [hunk 1]
```diff
@@ -105,8 +125,10 @@ class OLP_SDK_DEPRECATED("Deprecated. Will be removed in 04.2020")
 };
 
 /**
- * @brief TokenProviderDefault - provides authentication tokens given
- * appropriate OLP user credentials using the default minimum token validity.
+ * @brief Provides the authentication tokens using the default minimum token
+ * validity.
+ *
+ * The HERE OLP user credentials should be valid valid.
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h]:
```suggestion
 * The HERE OLP user credentials should be valid.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h [hunk 2]
```diff
@@ -80,8 +96,12 @@ class OLP_SDK_DEPRECATED("Deprecated. Will be removed in 04.2020")
   }
 
   /**
-   * @brief Allows the http status code of the last request to be retrieved.
-   * @returns http status code
+   * @brief Gets the HTTP status code of the last request.
+   * Return the HTTP code of the last token request if it was successful.
+   * Otherwise, returns the HTTP 503 Service Unavailable server error.
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h]: What's the point of duplicating the same info in the @returns statement? Is is required by HERE docs?

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h [hunk 3]
```diff
@@ -28,57 +28,70 @@
...
-   * @brief Constructor
+   * @brief Creates the default `TokenResult` instance.
    */
   TokenResult() = default;
 
   /**
-   * @brief Copy Constructor
-   * @param other the object to copy from
+   * @brief Creates the `TokenResult` instance that is a copy
+   * of the `other` parameter.
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h]:
```suggestion
   * of the `other` token result.
```
I think parameter is a bit generalized

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h [hunk 4]
```diff
@@ -28,57 +28,70 @@
...
 
   /**
-   * @brief Status code of the HTTP response, if a positive value returned. A
-   * negative value indicates a possible networking issue (retrying the request
-   * is recommended).
-   * @return Status code of the HTTP response
+   * @brief Gets the HTTP status code.
+   *
+   * @return The status code of the HTTP response if a positive value is
+   * returned. A negative value indicates a possible networking issue. If you
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h]:
```suggestion
   * returned. A negative value indicates a possible networking error. If you
```
I think issue is too generic.

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h [hunk 5]
```diff
@@ -27,68 +27,96 @@
...
+   * @param key Your access key ID.
+   * @param secret Your access key secret.
    */
   AuthenticationCredentials(std::string key, std::string secret);
 
   /**
-   * @brief Copy Constructor
-   * @param other object to copy from
+   * @brief Creates the `AuthenticationCredentials` instance that is a copy
+   * of the `other` parameter.
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h]:
```suggestion
   * of the `other` credentials.
```
Reason - the parameter is too generic

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h [hunk 6]
```diff
@@ -26,30 +26,43 @@
 
 namespace olp {
 namespace authentication {
+
+/**
+ * @brief Contains information on the authentication error.
+ *
+ * You can get the following information on the authentication error: the error
+ * code and error message.
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h]:
```suggestion
 * code (\ref GetErrorCode ) and error message ( \ref GetMessage ).
```
Maybe better add links to the accessors of those information?

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h [hunk 7]
```diff
@@ -26,30 +26,43 @@
...
+ */
 class AUTHENTICATION_API AuthenticationError {
  public:
   /**
-   * @brief Authentication error information container
+   * @brief Creates the default `AuthenticationError` instance.
    */
   AuthenticationError() = default;
 
   AuthenticationError(int network_code, const std::string& message)
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h]: Doxygen is missing

File: olp-cpp-sdk-authentication/include/olp/authentication/SignInUserResult.h [hunk 8]
```diff
@@ -31,63 +31,66 @@ namespace authentication {
...
- * operation.
+ * @brief A response to your sign-in operation.
  *
- * If a HTTP status of 412 (PRECONDITION_FAILED) or 201 (CREATED) is returned,
- * the response will return terms of re-acceptance. This response may be
- * received on the first sign in, or any subsequent sign-in, for any user
- * authentication method, as long as the user has not accepted the latest terms
- * for their country.
+ * If the HTTP status of 412 (PRECONDITION_FAILED) or 201 (CREATED) is returned,
+ * the response returns terms of re-acceptance. Not to receive this response
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/SignInUserResult.h]:
```suggestion
 * the response returns terms of re-acceptance. In order to not to receive this response
```
I don't know, your variant sounds a bit unusual for me (but maybe it is correct :) )

File: olp-cpp-sdk-authentication/include/olp/authentication/SignOutResult.h [hunk 9]
```diff
@@ -30,38 +30,44 @@ namespace authentication {
...
 
   /**
-   * @brief HTTP status code getter method.
-   * @return Status code of the HTTP response, if a positive value returned. A
-   * negative value indicates a possible networking issue (retrying the request
-   * is recommended).
+   * @brief Gets the HTTP status code.
+   *
+   * @return The status code of the HTTP response if a positive value is
+   * returned. A negative value indicates a possible networking issue. If you
```
Comment (borzun) [inline olp-cpp-sdk-authentication/include/olp/authentication/SignOutResult.h]:
```suggestion
   * returned. A negative value indicates a possible networking error. If you
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h (hunk 5 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h]: Disagree: `other` is not the name of the credentials. It's the name of the parameter that contains the credentials.

File: olp-cpp-sdk-authentication/include/olp/authentication/SignInUserResult.h (hunk 8 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/SignInUserResult.h]: "Not to" is a shorter version of "in order not to". Same as "to" is a shorer version of "in order to": *In order to/to* log in to the system, you have to provide your user credentials.

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h (hunk 1 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h]: Fixed.

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h (hunk 2 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h]: Deleted from the `brief`

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h (hunk 3 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h]: Fixed.

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h (hunk 4 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/TokenResult.h]: Changed.

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h (hunk 5 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h]: Changed to "the `other` parameter that contains the access credentials".

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h (hunk 7 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h]: Added.

File: olp-cpp-sdk-authentication/include/olp/authentication/SignOutResult.h (hunk 9 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/SignOutResult.h]: Fixed.

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h (hunk 6 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationError.h]: Added (here and in the other files).

**Pull request merged**